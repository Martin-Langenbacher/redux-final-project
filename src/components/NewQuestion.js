import { useState } from "react";
import { connect, useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";

/*

import { connect, useDispatch } from "react-redux";

import PollResults from "./PollResults";
import { handleSaveQuestionAnswer } from "../actions/users";

const PollPage = (props) => {
  const { id } = useParams();
  const dispatch = useDispatch();
*/

import { handleAddQuestion } from "../actions/questions";
import { addQuestionToUser } from "../actions/users";

//const NewQuestion = ( { dispatch }) => {
const NewQuestion = (props) => {
  const navigate = useNavigate();
  const dispatch = useDispatch();

  const maxLengthOfText = 50;
  const startToShow = 25;
  const [optionOneText, setOptionOneText] = useState("");
  const [optionTwoText, setOptionTwoText] = useState("");

  const handleChangeOptionOne = (e1) => {
    const textOptionOne = e1.target.value;
    setOptionOneText(textOptionOne);
  };
  const handleChangeOptionTwo = (e2) => {
    const textOptionTwo = e2.target.value;
    setOptionTwoText(textOptionTwo);
  };

  const handleSubmit = (submitEvent) => {
    submitEvent.preventDefault();

    dispatch(handleAddQuestion(optionOneText, optionTwoText)).then((qid) => {
      dispatch(addQuestionToUser(qid, props.authUser));
    });

    setOptionOneText("");
    setOptionTwoText("");

    navigate("/");
  };

  const questionTextOneLeft = maxLengthOfText - optionOneText.length;
  const questionTextTwoLeft = maxLengthOfText - optionTwoText.length;

  return (
    <div>
      <div className="center">
        <h2>Would You Rather</h2>
        <h4>Create Your Own Poll</h4>
      </div>
      <form className="new-question" onSubmit={handleSubmit}>
        <h5>First Option</h5>
        <textarea
          placeholder="Option One"
          value={optionOneText}
          onChange={handleChangeOptionOne}
          className="textarea"
          maxLength={maxLengthOfText}
        />
        {questionTextOneLeft <= startToShow && (
          <div className="question-length">{questionTextOneLeft}</div>
        )}

        <h5>Second Option</h5>
        <textarea
          placeholder="Option Two"
          value={optionTwoText}
          onChange={handleChangeOptionTwo}
          className="textarea"
          maxLength={maxLengthOfText}
        />
        {questionTextTwoLeft <= startToShow && (
          <div className="question-length">{questionTextTwoLeft}</div>
        )}
        <button
          className="btn"
          type="submit"
          disabled={optionOneText === "" || optionTwoText === ""}
        >
          Submit
        </button>
      </form>
    </div>
  );
};

const mapStateToProps = ({ authedUser, questions }) => ({
  authUser: authedUser,
  questions,
});

export default connect(mapStateToProps)(NewQuestion);

/*
const mapStateToProps = ({ authedUser, users, questions }) => {
  return {
    authUser: authedUser,
    userId: Object.keys(users),
    avatar: users[authedUser],
    questions: questions,
    users,
  };
};

*/

//export default connect()(NewQuestion);
