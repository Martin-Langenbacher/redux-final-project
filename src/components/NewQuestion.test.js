// INFORMATION
// press 'u', if the view really changed ! --> You get a new snapshot!

// is present:  ==> getBy
// not present: ==> queryBy

import * as React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

import NewQuestion from "./NewQuestion";

const mockStore = configureStore();

describe("NewQuestion", () => {
  it("matches the snapshot, no data is transfered", () => {
    const store = mockStore({});
    var view = render(
      <Provider store={store}>
        <Router>
          <NewQuestion />
        </Router>
      </Provider>
    );
    expect(view).toMatchSnapshot();
  });
});
