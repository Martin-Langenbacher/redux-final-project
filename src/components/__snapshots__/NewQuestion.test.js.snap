// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewQuestion matches the snapshot, no data is transfered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="center"
        >
          <h2>
            Would You Rather
          </h2>
          <h4>
            Create Your Own Poll
          </h4>
        </div>
        <form
          class="new-question"
        >
          <h5>
            First Option
          </h5>
          <textarea
            class="textarea"
            maxlength="50"
            placeholder="Option One"
          />
          <h5>
            Second Option
          </h5>
          <textarea
            class="textarea"
            maxlength="50"
            placeholder="Option Two"
          />
          <button
            class="btn"
            disabled=""
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="center"
      >
        <h2>
          Would You Rather
        </h2>
        <h4>
          Create Your Own Poll
        </h4>
      </div>
      <form
        class="new-question"
      >
        <h5>
          First Option
        </h5>
        <textarea
          class="textarea"
          maxlength="50"
          placeholder="Option One"
        />
        <h5>
          Second Option
        </h5>
        <textarea
          class="textarea"
          maxlength="50"
          placeholder="Option Two"
        />
        <button
          class="btn"
          disabled=""
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
